src/libmongoc/tests/unified/bson-parser.h

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2020-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "bson/bson.h"
+
+/* bson_parser_t is a very simplified parser to parse BSON fields into C values.
+ * Example usage:
+ *
+ * typedef struct {
+ *    char *name;
+ *    bool *has_kids;
+ *    bson_t *jobs;
+ * } person_t;
+ *
+ * person_t person;
+ * bson_parser_t *parser;
+ *
+ * parser = bson_parser_new ();
+ * bson_parser_utf8 (parser, "name", &person.name);
+ * bson_parser_bool_optional (parser, "hasKids", &person.has_kids);
+ * bson_parser_arr (parser, "jobHistory", &person.job_history);
+ * bson_parser_doc_alternate (parser, "jobs", &person.job_history);
+ * bson_parser_parse_or_assert (parser, &bson);
```

Comment:
`bson` is not defined yet. Did you mean to define it above? Or maybe move the comment above line 31 so it's clear which BSON document this code is parsing?